src/pages/iou/IOUCurrencySelection.js

```diff
@@ -56,125 +56,98 @@ const defaultProps = {
-     */
-    changeSearchValue(searchValue) {
-        const currencyOptions = this.getCurrencyOptions(this.props.currencyList);
-        const searchRegex = new RegExp(Str.escapeForRegExp(searchValue), 'i');
-        const filteredCurrencies = _.filter(currencyOptions, (currencyOption) => searchRegex.test(currencyOption.text));
-
-        this.setState({
-            searchValue,
-            currencyData: filteredCurrencies,
-        });
-    }
-
-    /**
-     * Confirms the selection of currency
-     *
-     * @param {Object} option
-     * @param {String} option.currencyCode
-     */
-    confirmCurrencySelection(option) {
-        const backTo = lodashGet(this.props.route, 'params.backTo', '');
-        // When we refresh the web, the money request route gets cleared from the navigation stack.
-        // Navigating to "backTo" will result in forward navigation instead, causing disruption to the currency selection.
-        // To prevent any negative experience, we have made the decision to simply close the currency selection page.
-        if (_.isEmpty(backTo) || this.props.navigation.getState().routes.length === 1) {
-            Navigation.goBack();
-        } else {
-            Navigation.navigate(`${this.props.route.params.backTo}?currency=${option.currencyCode}`);
-        }
-    }
-
-    render() {
-        const headerMessage = this.state.searchValue.trim() && !this.state.currencyData.length ? this.props.translate('common.noResultsFound') : '';
-        return (
-            <ScreenWrapper includeSafeAreaPaddingBottom={false}>
-                {({safeAreaPaddingBottomStyle}) => (
-                    <>
-                        <HeaderWithBackButton
-                            title={this.props.translate('iOUCurrencySelection.selectCurrency')}
-                            onBackButtonPress={() => Navigation.goBack(ROUTES.getIouRequestRoute(Navigation.getTopmostReportId()))}
-                        />
-                        <OptionsSelector
-                            sections={this.getSections()}
-                            onSelectRow={this.confirmCurrencySelection}
-                            value={this.state.searchValue}
-                            onChangeText={this.changeSearchValue}
-                            textInputLabel={this.props.translate('common.search')}
-                            headerMessage={headerMessage}
-                            safeAreaPaddingBottomStyle={safeAreaPaddingBottomStyle}
-                            initiallyFocusedOptionKey={_.get(
-                                _.find(this.state.currencyData, (currency) => currency.currencyCode === this.getSelectedCurrencyCode()),
-                                'keyForList',
-                            )}
-                            shouldHaveOptionSeparator
-                        />
-                    </>
-                )}
-            </ScreenWrapper>
-        );
-    }
+    }, [searchValue, currencyData, props.translate]);
```

Comment:
Surely it's better to just pass `props.translate` rather than `props`?

Reply:
for sure - props.translate should be in deps

Reply:
Thanks. I've read up on prop destructuring, is this how it's supposed to be done? 

`const { translate } = props;`

Reply:
I think I've done this correctly: https://github.com/Expensify/App/pull/20579/files#diff-496700da68574808c5981ac4884e39d73ea02df4aa59bd5869e0c9227defe4c2R94